pkg/secrets/s3.go

```diff
@@ -67,7 +67,7 @@ func parseS3SecretConfig(secretConfig string) (S3Secret, error) {
 		case "k":
```

Comment:
Same comment as for vault. Looks like we're parsing the token twice and repeat the code too. I'd expect `ParseToken()` to just return a ```{
engine string
params map[string]string
}```